teach.md

```diff
@@ -0,0 +1,217 @@
+> works really well for programming in the small in CS1 without all the
+> drawbacks I feel exist with dynamically typed languages.”
+> 
+> Mark Lewis, Trinity University, USA.
+
+> “The familiar yet flexible expression-oriented syntax, advanced support for pattern-matching, extensive collections library, powerful type system, and general no-nonsense approach to essential programming constructs are all game changers.”
+> 
+> Lionel Parreaux, Hong Kong University of Science and Technology, Hong Kong.
+
+Read over to learn good reasons to use Scala to teach programming, to know
+where in the world Scala is taught, and to discover teaching tools and materials
+used at various levels (from high-school to university).
+
+## Why Teach Scala
+
+### Multiple Paradigms
+
+While we learn new concepts we compare and contrast different but similar
+concept and relate to what we already know. This is a very important learning
+principle and many pedagogical strategies build on contrasting and comparing
+while exploring similarities and differences between concepts and ideas. Scala
+is an excellent language in supporting this pedagogical principle as it
+includes a pragmatic mix of concepts and ideas, ranging from simple to
+advanced, that can be contrasted and compared, step by step.
+
+> “One example I use in my teaching is contrasting different ways of solving the
+> configuration problem: early on we contrast a global mutable variable versus
+> default arguments and later when we get more advanced we contrast default
+> arguments with context parameters. This way we can get deep into the more
+> general discussion of the tradeoff between flexibility and complexity.”
+> 
+> Björn Regnell, Lund University, Sweden.
+
+> “We teach how to implement basic interpreters and type checkers. For this
+> purpose, algebraic data types with pattern matching and immutable data
+> structures, provided by functional languages, are essential.”
+> 
+> Jaemin Hong, KAIST, South Korea.
+
+> “Scala offers a wide variety of programming language constructs. By
+> integrating both OOP and functional aspects into one language, it is a great
+> tool to teach both paradigms and compare solutions within the same language.
+> At the same time, Scala is a great tool to teach software design: it includes
+> a whole spectrum of features ranging from basic constructs to advanced program
+> structuring techniques. This allows us to use the same language to discuss
+> various strategies to decompose a software into modules.”
+> 
+> Jonathan Brachthäuser, University of Tübingen, Germany.
+
+### Expressive
+
+Scala has a concise syntax that allows developers to focus on the intent of a
+program without being distracted by syntactic noise. Here is a “Hello world”
+program in Scala:
+
+~~~
+@main def run() = println("Hello, World!")
+~~~
+
+> “Scala is a fantastic teaching language for illustrating solutions with
+> different tradeoffs between the least complex concepts that does the job,
```

Comment:
```suggestion
> different tradeoffs between the least complex concept that does the job,
```

Reply:
Really?

"Between the concept" does not sound correct to me.

Reply:
Really?
"Between the concept" does not sound correct to me.

Reply:
Done in 55dfe20f

Reply:
> Really? "Between the concept" does not sound correct to me.

It's definitely off as well, but the remaining of the sentence really wanted "concept" to be singular. Also "the least" kind of calls for singular, although that's not always the case. "between" is somewhat off in the sentence, but it's a quote, so we can't fix it "too much".